src/InputFieldDescriptor.php

```diff
@@ -0,0 +1,345 @@
+        $this->targetPropertyOnSource = $targetPropertyOnSource;
+//        $this->magicProperty = null;
+    }
+
+//    public function setMagicProperty(string $magicProperty): void
+//    {
+//        if ($this->originalResolver !== null) {
+//            throw new GraphQLRuntimeException('You cannot modify the target method via setMagicProperty because it was already used. You can still wrap the callable using getResolver/setResolver');
+//        }
+//        $this->callable = null;
+//        $this->targetMethodOnSource = null;
+//        $this->targetPropertyOnSource = null;
+//        $this->magicProperty = $magicProperty;
+//    }
+
+    public function isInjectSource(): bool
+    {
+        return $this->injectSource;
+    }
+
+    public function setInjectSource(bool $injectSource): void
+    {
+        $this->injectSource = $injectSource;
+    }
+
+    public function getComment(): ?string
+    {
+        return $this->comment;
+    }
+
+    public function setComment(?string $comment): void
+    {
+        $this->comment = $comment;
+    }
+
+//    public function getDeprecationReason(): ?string
+//    {
+//        return $this->deprecationReason;
+//    }
+//
+//    public function setDeprecationReason(?string $deprecationReason): void
+//    {
+//        $this->deprecationReason = $deprecationReason;
+//    }
+
+    public function getMiddlewareAnnotations(): MiddlewareAnnotations
+    {
+        return $this->middlewareAnnotations;
+    }
+
+    public function setMiddlewareAnnotations(MiddlewareAnnotations $middlewareAnnotations): void
+    {
+        $this->middlewareAnnotations = $middlewareAnnotations;
+    }
+
+    public function getRefMethod(): ReflectionMethod
+    {
+        return $this->refMethod;
+    }
+
+    public function setRefMethod(ReflectionMethod $refMethod): void
+    {
+        $this->refMethod = $refMethod;
+    }
+
+    public function getRefProperty(): ReflectionProperty
+    {
+        return $this->refProperty;
+    }
+
+    public function setRefProperty(ReflectionProperty $refProperty): void
+    {
+        $this->refProperty = $refProperty;
+    }
+
+    /**
+     * Returns the original callable that will be used to resolve the field.
+     */
+    public function getOriginalResolver(): ResolverInterface
+    {
+        if (isset($this->originalResolver)) {
+            return $this->originalResolver;
+        }
+
+        if (is_callable($this->callable)) {
+            /** @var callable&array{0:object, 1:string} $callable */
+            $callable = $this->callable;
+            $this->originalResolver = new ServiceResolver($callable);
+        } elseif ($this->targetMethodOnSource !== null) {
+            $this->originalResolver = new SourceResolver($this->targetMethodOnSource);
+        } elseif ($this->targetPropertyOnSource !== null) {
+            $this->originalResolver = new SourceInputPropertyResolver($this->targetPropertyOnSource);
+//        } elseif ($this->magicProperty !== null) {
```

Comment:
Unused code

Reply:
the magic property code is to be enabled in a future PR

Reply:
Let's add a comment with any commented out code. In general, we should remove virtually all commented out code, but small lines like this are fine if we can share with others why it's there.